### PR TITLE
Add a proper constructor to GTK Closures.

### PR DESCRIPTION
**Pull request by hugopl**
They use the same mechanism used on signals.

### CODE REVIEW

**Comment by hugopl** [comment]
Future improvement: Would be nice to have all methods receiving a closure to (also) accept a block, so the user wouldn't need to explicit create the Closure object themselves.

File: samples/gtk_accel_group.cr [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+window.connect "destroy", ->Gtk.main_quit
+window.add Gtk::Label.new("Press Ctrl+S")
+
+accel_group = Gtk::AccelGroup.new
+# Passing a block
+closure = GObject::Closure.new do
+  puts "Ctrl+S pressed!"
+end
+
+accel_group.connect(Gdk::KEY_S, Gdk::ModifierType::CONTROL_MASK, Gtk::AccelFlags::ZERO_NONE, closure)
```
Comment (jhass) [inline samples/gtk_accel_group.cr:13]:
```suggestion
accel_group.connect(Gdk::KEY_S, :control_mask, :zero_none, closure)
```

I didn't test this but given Crystal's symbol to enum conversion I think that should work :)

Comment (hugopl) [reply samples/gtk_accel_group.cr:13]: It worked, nice. I didn't know about this language feature. I'll update the PR.

**Comment by jhass** [comment]
> Future improvement: Would be nice to have all methods receiving a closure to (also) accept a block, so the user wouldn't need to explicit create the Closure object themselves.

Yes this shouldn't be too hard and very similar to the `GValue` convenience stuff. https://github.com/jhass/crystal-gobject/blob/master/src/g_i_repository/info/function_info.cr#L76-L84

Basically there if there's a single arg of the `GObject::Closure` type, replace it with `&#{arg.name}` and then in the method body do `#{arg.name} = GObject::Closure.new(#{arg.name}` before the lib call.

**Comment by jhass** [comment]
🎉

**Comment by jhass** [comment]
Alright, I couldn't resist and pushed a commit generating block taking overloads :)

**Pull request merged**